homeassistant/components/husqvarna_automower/api.py

```diff
@@ -0,0 +1,30 @@
+"""API for Husqvarna Automower bound to Home Assistant OAuth."""
+
+import logging
+
+from aioautomower.auth import AbstractAuth
+from aioautomower.const import API_BASE_URL
+from aiohttp import ClientSession
+
+from homeassistant.helpers import config_entry_oauth2_flow
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AsyncConfigEntryAuth(AbstractAuth):
+    """Provide Husqvarna Automower authentication tied to an OAuth2 based config entry."""
+
+    def __init__(
+        self,
+        websession: ClientSession,
+        oauth_session: config_entry_oauth2_flow.OAuth2Session,
+    ) -> None:
+        """Initialize Husqvarna Automower auth."""
+        super().__init__(websession, API_BASE_URL)
+        self._oauth_session = oauth_session
+
+    async def async_get_access_token(self) -> str:
+        """Return a valid access token."""
+        if not self._oauth_session.valid_token:
+            await self._oauth_session.async_ensure_token_valid()
```

Comment:
```suggestion
        await self._oauth_session.async_ensure_token_valid()
```

https://github.com/home-assistant/core/blob/a547a4ecd53a56a463ff34b3a17cf952273f57f5/homeassistant/helpers/config_entry_oauth2_flow.py#L510-L513